### DISCUSSION TITLE
Should an input field with a null value use the default value just as if the field weren't included?

### DISCUSSION THREAD

**Discussion by andersonbd1** [Netflix/dgs-framework#1833] [General]
Hi,
I have this field declared in my graphql:
```   
myField: Boolean = false
```

I suppose I could see reasons for a contrary opinion on this, but I would expect the value to be set to false (not null) if the client explicitly passes in a null value.  This is not the case (at least in v5.4.1).  Instead the value is set to null.  I have a couple of question:

1. Should an input field with a null value use the default value just as if the field weren't included in the input?
2. If not, is there a way to configure this behavior (either across the entire application or explicitly on particular fields)?
3. If so, I think I found where I'd [change the code for v5.4.1](https://github.com/Netflix/dgs-framework/blob/v5.4.1/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultInputObjectMapper.kt#L98)`^1`, but how does this map to a branch?  I haven't done a ton of open source code contributions, but when I have I'd expect to see a v5 branch.

`^1`
change:
```
            if (declaredField != null) {
```
to
```
            if (declaredField != null && it.value != null) {
```

**Comment by srinivasankavitha** [comment]
Hi @andersonbd1 - This behavior is per spec as described here: https://spec.graphql.org/June2018/#sec-Type-System.Non-Null
The same applies for Boolean fields as well. If you want to ensure a default value is set, that would have to be specified in the schema with a default value for that input field.

**Reply by andersonbd1** [reply]
Thanks, @srinivasankavitha .  I don't however, see default values mentioned in the section you referenced?  Could you point me to the exact location?

> If you want to ensure a default value is set, that would have to be specified in the schema with a default value for that input field.

I have done that.  Where my assumption proved to be wrong is that when the client explicitly passes in a null value, then my default value isn't used.

**Reply by srinivasankavitha** [reply]
If the client explicitly sets null, then the same is what will be used, especially since it is a nullable field.

**Reply by andersonbd1** [reply]
Thanks.  If I made the field not-nullable and the client did not pass in the field, then would the default be used or would it be an error?

**Reply by srinivasankavitha** [reply]
It would use the default value in this case. See this section for the spec: https://spec.graphql.org/June2018/#sec-Required-Arguments

Hope that helps.

**Reply by andersonbd1** [reply]
Yes, that helps.  Thanks, @srinivasankavitha

**Reply by andersonbd1** [reply]
I just tested v5.4.1 and that is not the case.  

I changed the definition to be:
```
    ...
    myField: Boolean! = false
    ...
```
and then sent this to the endpoint:
```
    ...
    "myField": null,
    ....
```

And I get this:
```
{
  "errors": [
    {
      "message": "Variable 'input' has an invalid value: Field 'myField' has coerced Null value for NonNull type 'Boolean!'",
      "locations": [
        {
          "line": 1,
          "column": 43
        }
      ],
      "extensions": {
        "classification": "ValidationError"
      }
    }
  ]
}
```

It seems there is not a way to define in the backend a guaranteed default/non-null value for a field no matter what the client sends.

**Reply by srinivasankavitha** [reply]
But in this case, you are infact passing the null value for `myField`  where it is non-nullable. If you want the default value, you will need to omit specifying `myField` in the input argument. The default value will be picked up in that case.

Btw - I am stating what the current behavior is with 8.x which is what is actively supported. The other versions are ithere purely to avoid breaking users, but we are not actively fixing or making changes in those versions.
